src/ioctls/device.rs

```diff
@@ -59,44 +59,75 @@ impl FromRawFd for DeviceFd {
 mod tests {
     use super::*;
     use ioctls::system::Kvm;
+    #[cfg(target_arch = "aarch64")]
+    use kvm_bindings::kvm_device_type_KVM_DEV_TYPE_ARM_VGIC_V2;
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    use kvm_bindings::kvm_device_type_KVM_DEV_TYPE_VFIO;
     use kvm_bindings::{
-        kvm_device_type_KVM_DEV_TYPE_ARM_VGIC_V3, kvm_device_type_KVM_DEV_TYPE_VFIO,
-        KVM_CREATE_DEVICE_TEST, KVM_DEV_VFIO_GROUP, KVM_DEV_VFIO_GROUP_ADD,
+        kvm_device_type_KVM_DEV_TYPE_ARM_VGIC_V3, KVM_CREATE_DEVICE_TEST, KVM_DEV_VFIO_GROUP,
+        KVM_DEV_VFIO_GROUP_ADD,
     };
 
     #[test]
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
     fn test_create_device() {
-        #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
-        use kvm_bindings::kvm_device_type_KVM_DEV_TYPE_FSL_MPIC_20;
-
         let kvm = Kvm::new().unwrap();
         let vm = kvm.create_vm().unwrap();
 
         let mut gic_device = kvm_bindings::kvm_create_device {
-            #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
             type_: kvm_device_type_KVM_DEV_TYPE_ARM_VGIC_V3,
-            #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
-            type_: kvm_device_type_KVM_DEV_TYPE_FSL_MPIC_20,
             fd: 0,
             flags: KVM_CREATE_DEVICE_TEST,
         };
         // This fails on x86_64 because there is no VGIC there.
-        // This fails on aarch64 as it does not use MPIC (MultiProcessor Interrupt Controller), it uses
-        // the VGIC.
         assert!(vm.create_device(&mut gic_device).is_err());
 
-        if cfg!(any(target_arch = "x86", target_arch = "x86_64")) {
-            gic_device.type_ = kvm_device_type_KVM_DEV_TYPE_VFIO;
-        } else if cfg!(any(target_arch = "arm", target_arch = "aarch64")) {
-            gic_device.type_ = kvm_device_type_KVM_DEV_TYPE_ARM_VGIC_V3;
-        }
+        gic_device.type_ = kvm_device_type_KVM_DEV_TYPE_VFIO;
+
         let device_fd = vm
             .create_device(&mut gic_device)
             .expect("Cannot create KVM device");
 
-        let raw_fd = unsafe { libc::dup(device_fd.as_raw_fd()) };
```

Comment:
The code-coverage decrease prompted me to check and find out that these 3 lines in this particular test were there to also test (as coverage drop suggests) `DeviceFd::as_raw_fd()` and `DeviceFd::from_raw_fd()`.

I suggest you either:

- bring back those lines in this test only and properly document them with comments explaining what their purpose is in the test,
or
- add another test that covers `DeviceFd::as_raw_fd()` and `DeviceFd::from_raw_fd()`.

Reply:
Aha, yes, `as_raw_fd()` and `from_raw_fd()` are implemented for `DeviceFd`. Although they were not needed in our test cases, they should be tested.
I prefer to cover them in this test.